src/pyfmi/fmi_util.pyx

```diff
@@ -1000,99 +1000,81 @@ cdef class DumpData:
+@cython.wraparound(False)
+def read_trajectory(file_name, int data_index, int file_position, int sizeof_type, int nbr_points, int nbr_variables):
+    cdef int i = 0
+    cdef unsigned long int offset
+    cdef unsigned long int start_point = data_index*sizeof_type
+    cdef unsigned long int end_point   = sizeof_type*(nbr_points*nbr_variables)
+    cdef unsigned long int interval    = sizeof_type*nbr_variables
+    cdef FILE* cfile
+    cdef np.ndarray[DTYPE_t, ndim=1] data
+    cdef DTYPE_t* data_ptr
+    cdef size_t sizeof_dtype = sizeof(DTYPE_t)
+
+    cfile = fopen(file_name, 'rb')
+    data = np.empty(nbr_points, dtype=DTYPE)
+    data_ptr = <DTYPE_t*>data.data
+
     
-    def __init__(self, model, np.ndarray[np.uint32_t, ndim=1, mode='c'] real_var_ref, 
-                              np.ndarray[np.uint32_t, ndim=1, mode='c'] int_var_ref, 
-                              np.ndarray[np.uint32_t, ndim=1, mode='c'] bool_var_ref, 
-                              np.ndarray[np.int32_t, ndim=1, mode='c']  data_order, int nvariables):
-        
-        self.real_var_ref = real_var_ref
-        self.int_var_ref  = int_var_ref
-        self.bool_var_ref = bool_var_ref
-        self.data_order   = data_order
-        self.model = model
-        self.nvariables = nvariables
-        self.cache = range(nvariables-1)
-        #self.data = range(nvariables-1)
-        self.data = np.empty(nvariables)
-        self._cache = np.empty(nvariables-1)
-        
-        self.rs = self.real_var_ref.size
-        self.ris = self.real_var_ref.size+self.int_var_ref.size
-        self.ribs = self.real_var_ref.size+self.int_var_ref.size+self.bool_var_ref.size
+    fseek(cfile, file_position, 0)
+    #for offset in range(start_point, end_point, interval):
+    for offset from start_point <= offset < end_point by interval:
+        fseek(cfile, file_position+offset, 0)
+        fread(<void*>(data_ptr + i), sizeof_dtype, 1, cfile)
+        i = i + 1
 
-    def save_point(self):
-            cdef int k,j,l
-            
-            cdef np.ndarray correct_order
-            cdef np.ndarray[double, ndim=1, mode='c'] r
-            cdef np.ndarray[int,    ndim=1, mode='c'] i
-            cdef np.ndarray b
+    fclose(cfile)
 
-            #Retrieves the time-point
-            r = self.model.get_real(self.real_var_ref)
-            i = self.model.get_integer(self.int_var_ref)
-            b = self.model.get_boolean(self.bool_var_ref)
-            
-            for k in range(self.rs):
-                self.data[k] = r[k]
-            for k in range(self.rs, self.ris):
-                self.data[k] = i[k-self.rs]
-            for k in range(self.ris, self.ribs):
-                self.data[k] = b[k-self.ris]
-            
-            return self._create_string(self.data)
+    return data
+
+@cython.boundscheck(False)
+@cython.wraparound(False)
+def read_name_list(file_name, int file_position, int sizeof_type, int nbr_variables, int max_length):
```

Comment:
I miss a docstring describing the function and the arguments.

Reply:
Fixed.